skopt/space/space.py

```diff
@@ -518,6 +518,9 @@ def __init__(self, dimensions):
         """
         self.dimensions = [check_dimension(dim) for dim in dimensions]
 
+        # Names of all the dimensions in the search-space.
+        self.dimension_names = [dim.name for dim in self.dimensions]
```

Comment:
This should be a property like `is_real` below so the doc string is accessible in the REPL

Reply:
Good point!